lib/FrontendTool/FrontendTool.cpp

```diff
@@ -973,6 +973,34 @@ static Optional<bool> emitSILAfterSILGen(CompilerInvocation &Invocation,
                   opts.EmitSortedSIL);
 }
 
+static bool serializeMSF(FrontendOptions &opts, SILModule *SM,
+                         ASTContext &Context, ModuleOrSourceFile MSF) {
+  const std::string &moduleOutputPath = opts.ModuleOutputPath;
+  if (moduleOutputPath.empty())
+    return Context.hadError();
+  SerializationOptions serializationOpts;
+  serializationOpts.OutputPath = moduleOutputPath.c_str();
+  serializationOpts.SerializeAllSIL = true;
+  serializationOpts.IsSIB = true;
+  serialize(MSF, serializationOpts, SM);
+  return Context.hadError();
+}
+
+static Optional<bool> emitSIBAfterSILGen(CompilerInvocation &Invocation,
```

Comment:
…and it's probably not worth separating from this function, again.

Reply:
Agreed, after introducing linkAllIfNeeded and serializeSIB.